tests/services/custom_fields/test_date_cf.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Test ISODate and EDTF Custom Field."""
+
+import pytest
+from marshmallow import Schema, ValidationError
+
+from invenio_records_resources.services.custom_fields import (
+    EDTFDateStringCF,
+    ISODateStringCF,
+)
+
+
+#
+# ISODateCF
+#
+def test_isodatestring_mapping():
+    cf = ISODateStringCF("name")
+    assert cf.mapping == {"type": "date"}
+
+
+def test_isodatestring_validate():
+    cf = ISODateStringCF("name")
+    schema = Schema.from_dict({"test": cf.field})()
+
+    assert schema.load({"test": "1999-10-27"}) == {"test": "1999-10-27"}
+    pytest.raises(ValidationError, schema.load, {"f": "2020/2021"})
+
+
+#
+# EDTFDateStringCF
+#
+def test_edtfdatestring_mapping():
+    cf = EDTFDateStringCF("name")
+    assert cf.mapping == {
+        "type": "object",
+        "properties": {
+            "date": {"type": "keyword"},
+            "date_range": {"type": "date_range"},
+        },
+    }
+
+
+def test_edtfdatestring_validate():
+    cf = EDTFDateStringCF("name")
+    schema = Schema.from_dict({"test": cf.field})()
```

Comment:
**shelvable**: it seems to me the tests are testing the edtf main code while this has been tested or as you mentioned before, we should reuse what was already implemented in rdm-records... Myabe we should actually test the dump/load in a record-oriented workflow I.e create/update/search...

Reply:
I did part of _i.e create/update/search..._, checking both ES, API and DB contents. But I fully agree that's manual testing... I dont think reusing code is a fast move because of different layers. I would open an issue for that.

Reply:
Added a test see `test_dumpers.py`